src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -280,7 +348,13 @@ let rec doc_exp ctx (E_aux (e, (l, annot)) as full_exp) =
       let d_args = List.map (doc_exp ctx) args in
       nest 2 (parens (flow (break 1) (d_id :: d_args)))
   | E_vector vals -> failwith "vector found"
-  | E_typ (typ, e) -> parens (separate space [doc_exp ctx e; colon; doc_typ ctx typ])
+  | E_typ (typ, e) -> (
+      match e with
+      | E_aux (E_assign _, _) -> doc_exp ctx e
```

Comment:
I might try and get rid of these silly unit type annotations on assignments when they're generated, as they affect several backends.